### DISCUSSION TITLE
Type hinting for default variants in composed component.

### DISCUSSION THREAD

**Discussion by VaibhavAcharya** [stitchesjs/stitches#804] [Help] [answered]
I have a `TextPrimitive` which I use to create all other text components.
It looks like this ↓
```js
export const TextPrimitive = css({
  textRendering: "optimizeSpeed",

  variants: {
    font: {
      heading: {
        fontFamily: "$heading",
      },
      body: {
        fontFamily: "$body",
      },
      code: {
        fontFamily: "$code",
      }
    },
    // more variants...
  }
});
```

Now I create new text components by using this primitive & setting default variants for it.
Like this ↓
```js
export const Paragraph = styled("p", TextPrimitive, {
  defaultVariants: {
    font: "body",
    weight: "medium",
    // etc...
  }
});
```

This is indeed working as expected but I am not getting any type hinting in the `defaultVariants` of `Paragraph` like I get in `TextPrimitive`.

Am I doing something wrong?

**Comment by peduarte** [comment, marked as answer]
Yes, we havent added that in.

We ran into some TypeScript limitations there. The composition is quite deep, and TS doesn't know what to do. 

We will keep trying
